sycl/source/detail/scheduler/commands.hpp

```diff
@@ -424,12 +463,16 @@ class ExecCGCommand : public Command {
   void printDot(std::ostream &Stream) const final;
   void emitInstrumentationData();
 
+  const std::unique_ptr<detail::CG> &getCG() const { return MCommandGroup; }
```

Comment:
Why return `const std::unique_ptr<detail::CG> &` and not just a `const detail::CG&`?

Reply:
It's only preference.

Reply:
Unless you have a reason to use `unique_ptr` in the call site, it isn't neccessary. The only reason I can see for this not to return `const detail::CG&` is if `MCommandGroup` can be `nullptr`.

Reply:
I reckon `MCommandGroup` can't ever be null.

Reply:
Changed to return of reference instead of `unique_ptr`.